connectors/migrations/20241219_backfill_intercom_data_source_folders.ts

```diff
@@ -0,0 +1,119 @@
+import { makeScript } from "scripts/helpers";
+
+import {
+  getDataSourceNodeMimeType,
+  getHelpCenterCollectionInternalId,
+  getHelpCenterInternalId,
+  getTeamsInternalId,
+} from "@connectors/connectors/intercom/lib/utils";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { concurrentExecutor } from "@connectors/lib/async_utils";
+import { upsertDataSourceFolder } from "@connectors/lib/data_sources";
+import {
+  IntercomCollection,
+  IntercomHelpCenter,
+  IntercomTeam,
+  IntercomWorkspace,
+} from "@connectors/lib/models/intercom";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+async function createFolderNodes() {
+  const connectors = await ConnectorResource.listByType("intercom", {});
+
+  for (const connector of connectors) {
+    const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+    // Create Teams folder
+    await upsertDataSourceFolder({
+      dataSourceConfig,
+      folderId: getTeamsInternalId(connector.id),
+      parents: [getTeamsInternalId(connector.id)],
+      title: "Teams",
+      mimeType: getDataSourceNodeMimeType("TEAMS_FOLDER"),
+    });
+
+    const teams = await IntercomTeam.findAll({
+      where: {
+        connectorId: connector.id,
+      },
+    });
+    // Create a team folder for each team
+    await concurrentExecutor(
+      teams,
+      async (team) => {
+        const teamInternalId = getTeamsInternalId(connector.id);
+        await upsertDataSourceFolder({
+          dataSourceConfig,
+          folderId: teamInternalId,
+          parents: [getTeamsInternalId(connector.id)],
```

Comment:
Should be the team followed by the teams folder (the one that contains all teams)